infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/SidecarServicesProvisioner.java

```diff
@@ -51,18 +51,20 @@ public SidecarServicesProvisioner(List<ChePluginEndpoint> containerEndpoints, St
    */
   public void provision(KubernetesEnvironment kubernetesEnvironment)
       throws InfrastructureException {
-    for (ChePluginEndpoint endpoint : endpoints) {
-      String serviceName = endpoint.getName();
-      Service service = createService(serviceName, podName, endpoint.getTargetPort());
+    for (ChePluginEndpoint endpoint : endpoints) {;
+      if (Boolean.parseBoolean(endpoint.getAttributes().getOrDefault("discoverable", "true"))) {
```

Comment:
Increase of nesting harder to read, so maybe next approach would be better:
```
if (!Boolean.parseBoolean(endpoint.getAttributes().getOrDefault("discoverable", "true"))) {
    continue;
}
```
Apart from that, reading all the operations needed to check whether the server should be discoverable or not make code way less clean and harder too read. Even though you don't like single-use private methods I recommend using it to make code cleaner.